packages/teams-js/src/public/serializable.interface.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Interface for objects that can be serialized and passed to the host
+ */
+export interface ISerializable {
+  /**
+   * @returns A serializable representation of the object, used for passing objects to the host.
+   */
+  serialize(): unknown;
```

Comment:
>unknown [](http://example.com/codeflow?start=15&length=7)

Should the return type of this be `string`? Or at least `SimpleType`

Reply:
We should talk about this. I think we could definitely require string here, but there's no problem with passing lowercase-o `object`s through postMessage as long as those objects only contain primitives or objects that only contain primitives.

If we require string here that will mean that anyone who wants to serialize an object will have to turn it into a string representation. For simple objects like AppId that is very straightforward, but for objects with a lot of properties it would be less straightforward.

What I have here was my attempt at a compromise between "forcing correctness" and "developer convenenience", but I can be swayed if you feel strongly.

Reply:
After some offline discussion, we decided to change the type to `string | object` to be as restrictive as possible. If someone really needs to turn an object into a boolean (?) in the future we can talk about it.